docs.md

```diff
@@ -47,39 +51,112 @@ A type that describes the callback signature to [addListener](#31---addlistener)
 type IInsetCallbackFunc = (inset: IInsets) => void;
 ```
 
-## 3.0 - Insets
+### 2.3 - IInsetConfiguration
 
-`Insets` is a static class that provides the APIs to retrieve and to listen for inset changes.
+A type that describes the configuration object used when creating new `Inset` provider instances. All fields are optional.
 
-### 3.1 - addListener
+Structure:
+
+```typescript
+interface IInsetConfiguration {
+  mask?: number;
+  includeRoundedCorners?: boolean;
+}
+```
+
+If `mask` is not set, the default will be `DISPLAY_CUTOUT | SYSTEM_BARS`.
+If `includeRoundedCorners` is not set, the default will be `true`.
+
+### 2.4 - InsetMask
+
+An enumeration of mask values. These correlate to [WindowInsetsCompat.Type](https://developer.android.com/reference/androidx/core/view/WindowInsetsCompat.Type) though the values may not match, they are mapped during runtime.
+
+Currently the following mask values are supported:
+
+- CAPTION_BAR
+- DISPLAY_CUTOUT
+- IME
+- MANDATORY_SYSTEM_GESTURES
+- NAVIGATION_BARS
+- STATUS_BARS
+- SYSTEM_BARS
+- SYSTEM_GESTURES
+- TAPPABLE_ELEMENT
+
+If the android SDK introduces any newer ones, feel free to open a PR.
+
+## 3.0 - Inset
+
+`Inset` is a class that provides the APIs to retrieve and to listen for inset changes. It serves as a provider object and wil hold the configuration state as well as a collection of inset listeners. It is valid to have more than one `Inset` object alive in your application, with different configurations though it will be recommended to keep the instance count low for performance reasons.
+
+Direct access to the `Inset` constructor is forbidden but an instance can be created via the static `create` method. If the `Inset` provider instance is no
+longer used, it should be freed via `Insets.free(inset)` which will clean up
+references to objects and allow them to be garbage collected, both in the webview
+and in the Java/Native runtime. Additionally if your application holds any references to listener functions, they should also be cleaned up.
+
+Note that the configuration parameters cannot be modified on an inset provider
+instance.
+
+### 3.1 - create
+
+Creates a new `Inset` provider object with the given configuration. If the configuration object is missing, default values are used as indicated in [IInsetConfiguration](#23---iinsetconfiguration).
+
+The returned `Inset` instance can be used to attach listeners on for inset
+update notifications. When the provider is no longer needed, it should be freed by calling `[free()](#32---free).
```

Comment:
``[free()](#32---free)`

Random `. Possibly missing the ending ` for code styling?

Reply:
I originally had it as a inline code block, but then changed it to a link to the API. Looks like I just missed the backtick.